src/internal/connector/onedrive/url_cache.go

```diff
@@ -0,0 +1,275 @@
+package onedrive
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+type itemProperties struct {
+	downloadURL string
+	isDeleted   bool
+}
+
+// urlCache caches download URLs for drive items
+type urlCache struct {
+	driveID string
+	// urlMap stores Item ID -> item property map
+	urlMap          map[string]itemProperties
```

Comment:
Building the map k:v description into the var name does a better job describing how to use it than a comment.  Also, naming a map `fooMap` is a bit redundant.
```suggestion
	idToProps          map[string]itemProperties
```